src/processors.ts

```diff
@@ -28,18 +28,12 @@ if (process.env.CI !== undefined) {
  * This is increasingly useful the more files there are in the repository.
  */
 const getPreProcessor =
-  (diffFileList: string[], staged: boolean) =>
+  (trackedFileSet: Set<string>, diffFileList: string[]) =>
   (text: string, filename: string) => {
-    let untrackedFileList = getUntrackedFileList(staged);
-    const shouldRefresh =
-      !diffFileList.includes(filename) && !untrackedFileList.includes(filename);
```

Comment:
This condition not only matches new files but actually all unchanged files, so instead of speeding things up this actually slows the entire lint process.